### PR TITLE
adding figma upload support

### PR DESCRIPTION
**Pull request by Shahnawaz-Sk**

### CODE REVIEW

**Review by arushsaxena1998 — APPROVED** [review]

File: src/commander/config.ts [hunk 1]
```diff
@@ -19,3 +20,13 @@ configStatic
     .action(async function(filepath, options) {
         createWebStaticConfig(filepath);
     })
+
+configFigma
+.name('config:create-figma')
```
Comment (pinanks) [inline src/commander/config.ts:25]: Can you please fix indentation here

File: src/lib/httpClient.ts [hunk 2]
```diff
@@ -136,4 +136,15 @@ export default class httpClient {
             }
         }, log)
     }
+
+    getFigmaFilesAndImages(requestBody: any, log: Logger) {
+        return this.request({
+            url: "https://stage-api.lambdatestinternal.com/visualui/1.0/uploadfigma",
```
Comment (pinanks) [inline src/lib/httpClient.ts:142]: No need for the entire url, just path is fine

File: src/lib/constants.ts [hunk 3]
```diff
@@ -235,5 +235,34 @@ export default {
         'iPhone XR': { os: 'ios', viewport: {width: 414, height: 896}},
         'iPhone XS': { os: 'ios', viewport: {width: 375, height: 812}},
         'iPhone XS Max': { os: 'ios', viewport: {width: 414, height: 896}},
+    },
+
+    FIGMA_API : 'https://api.figma.com/v1/',
+    LSRS_API: 'https://stage-api.lambdatestinternal.com/visualui/1.0/screenshot/save',
```
Comment (pinanks) [inline src/lib/constants.ts:241]: No need for this.

File: src/lib/uploadFigmaDesigns.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import chalk from "chalk";
+import { mapIdToName } from "./utils.js";
+import constants from "./constants.js";
+
+export default async (ctx: Context): Promise<string> => { // Return type changed to string[]
+  const figmaConfigs = ctx.figmaDesignConfig.figma_config;
+  let results = "";
+
+  for (const config of figmaConfigs) {
+    const figmaToken = process.env.FIGMA_TOKEN;
```
Comment (pinanks) [inline src/lib/uploadFigmaDesigns.ts:11]: add all env variables in the env interface in context

File: src/lib/uploadFigmaDesigns.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      queryParams += `&ids=${fileIds}`;
+    }
+
+    let authToken = "";
+    if (process.env.LT_USERNAME != null && process.env.LT_ACCESS_KEY != null) {
+      authToken = `Basic ${Buffer.from(`${process.env.LT_USERNAME}:${process.env.LT_ACCESS_KEY}`).toString("base64")}`
+    }
+
+    // Construct the request body
+    const requestBody = {
```
Comment (pinanks) [inline src/lib/uploadFigmaDesigns.ts:25]: pass necessary variables to client function and create the request body there. Also, project token is added to every reqeust so need for it again.

File: src/lib/uploadFigmaDesigns.ts [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+    figmaFileToken = config.figma_file_token;
+    let queryParams = "";
+    if (config.figma_ids && config.figma_ids.length > 0) {
+      const fileIds = config.figma_ids.join(",");
+      queryParams += `&ids=${fileIds}`;
+    }
+
+    let authToken = "";
+    if (process.env.LT_USERNAME != null && process.env.LT_ACCESS_KEY != null) {
```
Comment (pinanks) [inline src/lib/uploadFigmaDesigns.ts:21]: env variables

**Review by pinanks — APPROVED** [review]

**Pull request merged**